src/makeInfinityAdder.js

```diff
@@ -6,7 +6,23 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  function adder(value) {
+    if (value === undefined) {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    } else {
+      sum += value;
+
+      return adder;
+    }
```

Comment:
else condition is redundant since you've used early return above
```suggestion
    }
    
    sum += value;

    return adder; 
```